downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile.java

```diff
@@ -0,0 +1,413 @@
+    // that are specific to each PubMed annotation will be output, or, if there are no literatureReferences just a single line with a Reactome identifier will be output.
+    // The GOA generation will be called differently depending on this.
+    private static List<String> getGOMolecularFunctionLine(GKInstance catalystInst, GKInstance referenceEntityInst, String taxonIdentifier, GKInstance reactionInst) throws Exception {
+        List<String> goAnnotationLines = new ArrayList<>();
+        List<String> pubMedIdentifiers = new ArrayList<>();
+        for (GKInstance literatureReferenceInst : (Collection<GKInstance>) catalystInst.getAttributeValuesList(ReactomeJavaConstants.literatureReference)) {
+            pubMedIdentifiers.add("PMID:" + literatureReferenceInst.getAttributeValue(ReactomeJavaConstants.pubMedIdentifier).toString());
+        }
+        GKInstance activityInst = (GKInstance) catalystInst.getAttributeValue(ReactomeJavaConstants.activity);
+        if (!activityInst.getAttributeValue(ReactomeJavaConstants.accession).toString().equals(PROTEIN_BINDING_ANNOTATION)) {
+            String goAccession = "GO:" + activityInst.getAttributeValue(ReactomeJavaConstants.accession).toString();
+            if (pubMedIdentifiers.size() > 0) {
+                for (String pubmedIdentifier : pubMedIdentifiers) {
+                    String goaLine = generateGOALine(referenceEntityInst, "F", goAccession, pubmedIdentifier, "EXP", taxonIdentifier);
+                    assignDateForGOALine(catalystInst, goaLine);
+                    goAnnotationLines.add(goaLine);
+                }
+            } else {
+                String reactomeIdentifier=  "REACTOME:" + ((GKInstance) reactionInst.getAttributeValue(ReactomeJavaConstants.stableIdentifier)).getAttributeValue(ReactomeJavaConstants.identifier).toString();
+                String goaLine = generateGOALine(referenceEntityInst, "F", goAccession, reactomeIdentifier, "TAS", taxonIdentifier);
+                assignDateForGOALine(catalystInst, goaLine);
+                goAnnotationLines.add(goaLine);
+            }
+        }
+        return goAnnotationLines;
+    }
+
+    // Before creating GOA line for BP annotations, the reaction in question needs to be checked for the existance of a 'goBiologicalProcess' attribute. If there is none
+    // than the instance's 'hasEvent' referrals are checked for any.
+    private static List<String> getGOBiologicalProcessLine(GKInstance referenceEntityInst, GKInstance reactionInst, String taxonIdentifier) throws Exception {
+        List<String> goAnnotationLines = new ArrayList<>();
+        for (Map<String, String> biologicalProcessAccession : getGOBiologicalProcessAccessions(reactionInst, 0)) {
+            String goaLine = generateGOALine(referenceEntityInst, "P", biologicalProcessAccession.get("accession"), biologicalProcessAccession.get("event"), "TAS", taxonIdentifier);
```

Comment:
Many calls to this function just use Strings as the 2nd and 5th argument. Things like "P", "F", "TAS", ... these are rather cryptic. The documentation for the function does not explain these arguments or their possible options. Would it be possible to clean this up with either:
 -  String constants, with more descriptive names for the values;
 -  OR using enumerations?

Reply:
Done, used a String constant